### ISSUE TITLE
Settings not saved?

### ISSUE THREAD

**Issue by pcace** [e-alfred/ocdownloader#162]
Hey there, 

i cannot save settings in OCdownloader (seed /time limit)
I am running debian 10, Nextcloud 18.0.3 and OCdownloader from git commit https://github.com/e-alfred/ocdownloader/commit/05d267ebc2044918310a5e24f7c642e8e6ea2d53

All files in apps/ocdownloader are owned by www-data /www-data. 
Where can i start to figure out what is happening? 

The behavior: i change the setting, click save and then i get a green "Settings Saved" sighn, but the settings reset to the value before. 

Any Idea what i can do there? 

Thank you so much,

Pcace

**Comment by e-alfred** [comment]
What type of database do you use? Do you get any error in the browser console or Nextcloud log?

**Comment by bro2020** [comment]
I have the same problem in Nextcloud 18.0.4. My installation:
Docker version 19.03.6, build 369ce74a3c
PHP 7.3.18 (cli) (built: May 15 2020 13:33:15) (NTS)
ENV PG_VERSION = 12.3

**Comment by e-alfred** [comment]
So you use Prostgresql? Any errors in any logs?

**Comment by bro2020** [comment]
Yes, docker postgres:alpine
There is nothing in the nextcloud log about this problem
And when downloading any video from YouTube, the error "Unable to retrieve true YouTube video URL"
But manually youtube-dl successfully downloads these videos

**Comment by bro2020** [comment]
I save user settings and such a request is sent to the database:

UPDATE oc_ocdownloader_personalsettings SET "VAL" = $1 WHERE "KEY" = $2 AND "UID" = $3

Settings not saved.

**Comment by bro2020** [comment]
![image](https://user-images.githubusercontent.com/65536615/87206125-b8c4ff80-c311-11ea-940a-04aa8a2e3b95.png)

The value was written to the table, but the Nextcloud interface reset to default...

![image](https://user-images.githubusercontent.com/65536615/87206290-15c0b580-c312-11ea-844d-8e40e028ddb3.png)

**Comment by MadMakz** [comment]
Same here since a while. Sttings not saved and youtube-dl broken as seen above.

NC 18.0.7 stable & MariaDB 10.1.45

**Comment by BorussiaPlayer17** [comment]
Still broken ? I’m using docker, nc 22.1 and Postgres on a pi4.

**Comment by aledr** [comment]
On NC 23.0.1 I do see an Internal Server Error while trying to access ocDownloader.
Please reach out if you want a debug session.

**Comment by weeman1337** [comment]
At anyone here: Tt would really help if you can check your logs for an error or exception and post it here. Thanks

**Comment by neozerosv** [comment]
I had the same problem, what I did to solve is:

1.  Add a espace  after the single quote and  before ` SET VAL`  and  ` SET val`   the file apps/ocdownloader/controller/lib/settings.php on lines 113,116 . 
2. Change the column names from lowercase to uppercase from the table oc_ocdownloader_queue with:
`\c nextcloud
alter table oc_ocdownloader_queue rename column status to "STATUS";
alter table oc_ocdownloader_queue rename column uid to "UID";
alter table oc_ocdownloader_queue rename column timestamp to "TIMESTAMP";
alter table oc_ocdownloader_queue rename column is_cleaned to "IS_CLEANED";
alter table oc_ocdownloader_queue rename column gid to "GID";
alter table oc_ocdownloader_queue rename column filename to "FILENAME";
alter table oc_ocdownloader_queue rename column protocol to "PROTOCOL";
\d  oc_ocdownloader_queue 
`

And thats how I got it working.
